File: jSKWDA/src/net/skwh/Edge.java
package net.skwh;

public class Edge {
	private Node c1;
	private Node c2;
	private int v;
	
	public Node getC1() {
		return c1;
	}
	public Node getC2() {
		return c2;
	}
	public int getV() {
		return v;
	}
	public Edge (Node C1, Node C2, int value) {
		c1 = C1;
		c2 = C2;
		v = value;
	}
}

File: cSKWDA/cSKWDA/Edge.cpp
#include "StdAfx.h"
#include "Edge.h"

Edge::Edge(Node N1, Node N2, int length) {
	*n1 = N1;
	*n2 = N2;
	v = length;
	n1->setAsEdge(this);
	n2->setAsEdge(this);
}
Edge::~Edge(void) {}

int Edge::getV() {
	return v;
}
void Edge::setV(int d) {
	v = d;
}
Node Edge::getN1() {
	return *n1;
}
void Edge::setN1(Node n) {
	*n1 = n;
}
Node Edge::getN2() {
	return *n2;
}
void Edge::setN2(Node n) {
	*n2 = n;
}